lib/Doctrine/Common/Collections/ArrayCollection.php

```diff
@@ -253,6 +255,12 @@ public function exists(Closure $p)
 
     /**
      * {@inheritDoc}
+     *
+     * @psalm-param TMaybeContained $element
+     *
+     * @psalm-return (TMaybeContained is T ? TKey|false : false)
```

Comment:
If I don't copy/paste this here, I get
```
ERROR: InvalidReturnType - lib/Doctrine/Common/Collections/ArrayCollection.php:261:21 - The declared return type '(TKey:Doctrine\Common\Collections\ReadableCollection as array-key)|false' for Doctrine\Common\Collections\ArrayCollection::indexOf is incorrect, got '(TKey:Doctrine\Common\Collections\ArrayCollection as array-key)|false' (see https://psalm.dev/011)
    public function indexOf($element)
```
from psalm

Reply:
I'd have to spend a bit more time looking to be sure, but I'm guessing that's probably https://github.com/vimeo/psalm/issues/7553. Docblock inheritance and templates don't play well with one another in Psalm, probably due to the way `TKey` on `ReadableCollection` is a different type from `TKey` on `ArrayCollection` and the docblock inheritance isn't automatically translating it based on the `@implements`/`@extends` on the class/interface declaration.